.github/workflows/ci.yml

```diff
@@ -135,7 +137,7 @@ jobs:
       uses: actions-rs/cargo@v1
       with:
         command: check
-        args: --all --bins --tests --examples
+        args: --workspace --all-features --bins --tests --examples --benches
```

Comment:
why having `--all-features` is good to test against, we also need to have tests (or at least build checks) for each individual feature, to make sure they work by themselves and with no features enabled at all

Reply:
Yes, that's true.  I was trying to take small steps.

Given how difficult it was to get this to be happy in CI I'd prefer to do that as a followup since it already fixes some things on main and would stop regressions to that instead of having to keep playing catchup with main while working on this - as already happened.

Reply:
this was not meant to add more work to this pr, just a reminder for all of us that it needs more work 😅

Reply:
Made an issue to not forget: https://github.com/n0-computer/iroh/issues/496